include/grub/gpt_partition.h

```diff
@@ -197,6 +190,34 @@ struct grub_gpt
 };
 typedef struct grub_gpt *grub_gpt_t;
 
+/* Helpers for checking the gpt status field.  */
+static inline int
+grub_gpt_mbr_valid (grub_gpt_t gpt)
+{
+  return (gpt->status & GRUB_GPT_PROTECTIVE_MBR ||
```

Comment:
I've made it a habit to always surround my bitwise operations in parens, since the precedence rules can be so subtle and hide bugs testing doesn't catch so well.  So these functions read suspiciously to me...